File: Client/src/app/shared/models/TicketSubtype.ts
export interface TicketSubtype{
    id: number,
    name: string,
    sequence: number,
    active: boolean,

    ticketType: TicketType,
}

export interface TicketType{
    id: number,
    name: string,
    sequence: number,
    active: boolean,
}

File: Backend/src/main/java/rc/bootsecurity/requestModule/ticketModule/entity/TicketSubtype.java
package rc.bootsecurity.requestModule.ticketModule.entity;

import lombok.Data;
import javax.persistence.*;

@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@Data
public abstract class TicketSubtype {
    @Id
    @GeneratedValue(strategy = GenerationType.TABLE)
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "sequence")
    private Integer sequence;

    @Column(name = "active")
    private Boolean active;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ticket_type_id")
    private TicketType ticketType;



}
